src/gui/entry/EntryAttachmentsWidget.h

```diff
@@ -0,0 +1,58 @@
+#ifndef ENTRYATTACHMENTSWIDGET_H
+#define ENTRYATTACHMENTSWIDGET_H
+
+#include <QWidget>
+
+namespace Ui {
+class EntryAttachmentsWidget;
+}
+
+class EntryAttachments;
+class EntryAttachmentsModel;
+
+class EntryAttachmentsWidget : public QWidget
+{
+    Q_OBJECT
+    Q_PROPERTY(bool readOnly READ isReadOnly WRITE setReadOnly NOTIFY readOnlyChanged)
+    Q_PROPERTY(bool isButtonsVisible READ isButtonsVisible WRITE setButtonsVisible NOTIFY buttonsVisibleChanged)
+public:
+    explicit EntryAttachmentsWidget(QWidget* parent = nullptr);
+    ~EntryAttachmentsWidget();
+
+    const EntryAttachments* entryAttachments() const;
+    bool isReadOnly() const;
+    bool isButtonsVisible() const;
+
+public slots:
+    void setEntryAttachments(const EntryAttachments* attachments);
+    void clearAttachments();
+    void setReadOnly(bool readOnly);
+    void setButtonsVisible(bool isButtonsVisible);
+
+signals:
+    void errorOccurred(const QString& error);
+    void readOnlyChanged(bool readOnly);
+    void buttonsVisibleChanged(bool isButtonsVisible);
+
+private slots:
+    void insertAttachments();
+    void removeSelectedAttachments();
+    void saveSelectedAttachments();
+    void openAttachment(const QModelIndex& index);
+    void openSelectedAttachments();
+    void updateButtonsEnabled();
+
+private:
+    bool insertAttachments(const QStringList& fileNames, QString* errorMessage);
+    bool openAttachment(const QModelIndex& index, QString* errorMessage);
+
+    bool eventFilter(QObject* watched, QEvent* event) override;
+
+    Ui::EntryAttachmentsWidget* m_ui;
+    EntryAttachments* m_entryAttachments;
+    EntryAttachmentsModel* const m_attachmentsModel;
```

Comment:
These two should be QPointers.

Reply:
I think that it is not necessary, they are only used internally (they are not re-assigned in the process of work)

Reply:
No, but it also doesn't hurt anybody and may catch errors further down the road. Who knows where these objects are used in 5 months...

Reply:
BTW I really only mean QPointers, not QSharedPointers or QScopedPointers. QPointers simply guard the pointer and reset to 0 if the QObject is deallocated. They don't count references or own anything and they can be converted to and from raw pointers.

Reply:
Done